evap/staff/importers.py

```diff
@@ -97,28 +105,111 @@ def store_in_database(self, vote_start_datetime, vote_end_date, semester):
+            evaluation = Evaluation(
+                vote_start_datetime=vote_start_datetime,
+                vote_end_date=vote_end_date,
+                course=course,
+                wait_for_grade_upload_before_publishing=self.is_graded,
+            )
+            evaluation.save()
+            evaluation.contributions.create(
+                evaluation=evaluation,
+                contributor=responsible_dbobj,
+                role=Contribution.Role.EDITOR,
+                textanswer_visibility=Contribution.TextAnswerVisibility.GENERAL_TEXTANSWERS,
+            )
+
+    def check_existing_course(self, semester, importer):
+        name_collision = self.check_name_collision(semester)
+        if not name_collision:
+            return False
+
+        if name_collision != NameCollision.BOTH:
+            importer.errors[ImporterError.COURSE].append(
+                _("Course {} ({}) does already exist in this semester.").format(
+                    self.name_en if name_collision == NameCollision.NAME_EN else self.name_de,
+                    "EN" if name_collision == NameCollision.NAME_EN else "DE",
+                )
+            )
+            return False
+
+        course = Course.objects.filter(
             semester=semester,
+            name_en=self.name_en,
+            name_de=self.name_de,
+        ).first()
+
+        if course is None:
+            # This can only happen if the name collisions happened in different courses
+            importer.errors[ImporterError.COURSE].append(
+                _("Course {} (EN) and Course {} (DE) do already exist separately in this semester.").format(
+                    self.name_en, self.name_de
+                )
+            )
+            return False
+
+        mismatches, has_wrong_evaluation_count = self.check_existing_course_attributes(course)
+
+        if mismatches or has_wrong_evaluation_count:
+            messages = (
+                [
+                    "the {} {}".format(
+                        " and ".join(mismatches),
+                        ngettext("does not match", "do not match", len(mismatches)),
+                    )
+                ]
+                if mismatches
+                else []
+            )
+            messages += ["the course must have exactly one evaluation"] if has_wrong_evaluation_count else []
+            importer.errors[ImporterError.COURSE].append(
+                _("Course {} ({}) does already exist in this semester but {}.").format(
+                    self.name_en,
+                    self.name_de,
+                    " and ".join(messages),
+                )
+            )
+        else:
+            self.existing_course = course
+
+        return self.existing_course is not None
+
+    def check_name_collision(self, semester):
+        flag = NameCollision.NONE
+        flag |= NameCollision(
+            Course.objects.filter(semester=semester, name_en=self.name_en).exists() and NameCollision.NAME_EN
         )
```

Comment:
Ah, this didn't work out like we first thought; I guess we should just have 
```suggestion
        flag = NameCollision.NONE
        if Course.objects.filter(semester=semester, name_en=self.name_en).exists():
            flag |= NameCollision.NAME_EN
```